core/tests/protractor_utils/CollectionEditorPage.js

```diff
@@ -49,123 +49,122 @@ var CollectionEditorPage = function() {
   var saveInProgressLabel = element(by.css(
     '.protractor-test-save-in-progress-label'));
 
-  this.addExistingExploration = function(explorationId) {
-    addExplorationInput.sendKeys(explorationId);
+  this.addExistingExploration = async function(explorationId) {
+    await waitFor.visibilityOf(
+      addExplorationInput, 'Add Exploration Input is not visible');
+    await addExplorationInput.sendKeys(explorationId);
     // Waits until the button becomes active after debouncing.
-    waitFor.elementToBeClickable(
+    await waitFor.elementToBeClickable(
       addExplorationButton,
       'Unable to find exploration ID: ' + explorationId);
-    addExplorationButton.click();
+    await addExplorationButton.click();
   };
 
   // Search and add existing exploration to the node graph.
-  this.searchForAndAddExistingExploration = function(query) {
-    waitFor.visibilityOf(
+  this.searchForAndAddExistingExploration = async function(query) {
+    await waitFor.visibilityOf(
       addExplorationInput, 'Add Exploration Input is not visible');
-    addExplorationInput.sendKeys(query);
+    await addExplorationInput.sendKeys(query);
     // Need to wait for result to appear.
-    waitFor.elementToBeClickable(
+    await waitFor.elementToBeClickable(
       addExplorationButton, 'Unable to find exploration: ' + query);
 
     var matched = false;
-    var dropdownResultElements = element.all(by.css('.dropdown-menu'));
-    dropdownResultElements.map(function(dropdownResult) {
-      return dropdownResult.getText();
-    }).then(function(listOfResult) {
-      listOfResult.forEach(function(element, index) {
-        if (element.indexOf(query) >= 0) {
-          // Selects the exploration from dropdown.
-          dropdownResultElements.get(index).click();
+    var dropdownResultElements = await element.all(by.css('.dropdown-menu'));
+    await dropdownResultElements.forEach(
```

Comment:
Instead of `await`ing an `element.all` call, use a `for` loop based on `await dropdownResultElements.count()`. This is preferable because `.forEach` is not reliable in the async-await paradigm

Alternatively, leave the `.map()` style that we used before

Reply:
Done.